File: ex10/ex10_04/ex10_04.js
const myObj = {
  name: "tclin",
  age: 68,
  getName: function() { return myObj.name; },
  setName: function(name) { myObj.name = name; },
  getAge: function() { return myObj.age; },
  setAge: function(age) { myObj.age = age; }   
};
function main() {
  myObj.hobbies = ["reading", "jogging", "coding"];
  myObj.hello = function () {
    return `Hello, ${myObj.name}!`;
  };
  console.log(myObj.name);
  console.log(myObj.hobbies);
  console.log(myObj.hello());
  console.log(myObj.getName());
  myObj.setAge(70);
  console.log(myObj.getAge());
  myObj.setName("John");
  console.log(myObj.getName());
}
main();

File: ex10/ex10_04/ex10_04.py
my_obj = {
  "name": "tclin",
  "age": 68,
  "get_age": lambda: my_obj["age"],
  "set_age": lambda age: my_obj.update({"age": age}),
  "get_name": lambda: my_obj["name"],
  "set_name": lambda name: my_obj.update({"name": name}),
}
def main():
  my_obj["hobbies"] = ["reading", "jogging", "coding"]
  my_obj["hello"] = lambda: f"Hello, {my_obj['name']}!"
  print(my_obj["name"])
  print(my_obj["hobbies"])
  print(my_obj["hello"]())
  print(my_obj["get_name"]())
  my_obj["set_age"](70)
  print(my_obj["get_age"]())
  my_obj["set_name"]("John")
  print(my_obj["get_name"]())
if __name__ == "__main__":
  main()